src/Stratis.SmartContracts.CLR.Tests/SmartContracts/SystemContractsDictionary.cs

```diff
@@ -0,0 +1,205 @@
+﻿using Stratis.SmartContracts;
+using ECRecover=Stratis.SCL.Crypto.ECRecover;
+
+public struct WhiteListEntry
+{
+    public UInt256 CodeHash;
+    public Address LastAddress;
+    public string Name;
+}
+
+public class SystemContractsDictionary : SmartContract
+{
+    const string primaryGroup = "main";
+
+    public SystemContractsDictionary(ISmartContractState state) : base(state)
+    {
+        this.State.SetArray($"Signatories:{primaryGroup}", new[] { new Address(0, 0, 0, 0, 0), new Address(0, 0, 0, 0, 1), new Address(0, 0, 0, 0, 2) });
```

Comment:
The "default" values should be updated at some stage with the correct default values. The `AddSignatory` and `RemoveSignatory` will allow the defaults to be changed. I am worried about passing the defaults to the constructor as maybe this will lead to reduced or single person security.

Reply:
State should be access by getters/setters in everywhere

Reply:
Done